zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -840,107 +913,79 @@ mod tests {
+                           transactions.block       AS block,
+                           transactions.txid        AS txid,
+                           ro.pool                  AS pool,
                            id_within_pool_table,
-                           v_received_notes.value          AS value,
+                           ro.value                 AS value,
                            CASE
-                                WHEN v_received_notes.is_change THEN 1
+                                WHEN ro.is_change THEN 1
                                 ELSE 0
                            END AS is_change,
                            CASE
-                                WHEN v_received_notes.is_change THEN 0
+                                WHEN ro.is_change THEN 0
                                 ELSE 1
                            END AS received_count,
                            CASE
-                             WHEN (v_received_notes.memo IS NULL OR v_received_notes.memo = X'F6')
+                             WHEN (ro.memo IS NULL OR ro.memo = X'F6')
                                THEN 0
                              ELSE 1
                            END AS memo_present
-                    FROM v_received_notes
+                    FROM v_received_outputs ro
                     JOIN transactions
-                         ON transactions.id_tx = v_received_notes.tx
+                         ON transactions.id_tx = ro.transaction_id
                     UNION
-                    -- Transparent TXOs received in this transaction
-                    SELECT utxos.received_by_account_id AS account_id,
-                           utxos.height                 AS block,
-                           utxos.prevout_txid           AS txid,
-                           0                            AS pool,
-                           utxos.id                     AS id_within_pool_table,
-                           utxos.value_zat              AS value,
-                           0                            AS is_change,
-                           1                            AS received_count,
-                           0                            AS memo_present
-                    FROM utxos
-                    UNION
-                    -- Shielded notes spent in this transaction
-                    SELECT v_received_notes.account_id  AS account_id,
-                           transactions.block           AS block,
-                           transactions.txid            AS txid,
-                           v_received_notes.pool        AS pool,
+                    -- Outputs spent in this transaction
+                    SELECT ro.account_id            AS account_id,
+                           transactions.block       AS block,
+                           transactions.txid        AS txid,
+                           ro.pool                  AS pool,
                            id_within_pool_table,
```

Comment:
This is hazardous because it mixes together ids from several tables.

Reply:
@str4d points out https://github.com/zcash/librustzcash/commit/8b8757ce659f981eb6a3e0aa69c27d9115f3a783 which was a fix to an even worse situation.

Reply:
We could avoid this if we were using UUIDs for primary keys (as has been the best practice for a lot of years now) but for here I don't think we can avoid this.

Reply:
It has a BLOB type that we could use to encode the pool and id. Not sure that is worth the complexity of encoding and decoding.

You'll love [this gem from the SQLite docs](https://www.sqlite.org/datatype3.html#affinity_name_examples):
> Note that a declared type of "FLOATING POINT" would give INTEGER affinity, not REAL affinity, due to the "INT" at the end of "POINT".

:sob:

Reply:
In the cold light of day, using BLOB sounds like a bad idea because you then either have to encode a pool field as well, which is a redundant encoding that could be inconsistent, or you have no way to obtain the pool for SQL queries. And there's no way of saying "pair for which it's okay to destructure the first element but not the second" in SQL.